app/views/subscriptions_management/index.html.erb

```diff
@@ -2,15 +2,30 @@
 
 <div class="govuk-grid-row">
   <div class="govuk-grid-column-two-thirds">
-    <h1 class="govuk-heading-l"><%= t("subscriptions_management.heading") %></h1>
-
     <% if flash[:success] %>
       <%= render 'govuk_publishing_components/components/success_alert', {
         message: flash[:success]["message"],
         description: flash[:success]["description"],
       } %>
     <% end %>
 
+    <% if flash[:subscription] %>
+      <% subscription = @subscriptions[flash[:subscription]['id']] %>
```

Comment:
Is there any risk that this will result in a nil? I'm assuming flash data gets set on a previous request and there's always a remove chance that `@subscriptions` doesn't have the key - which, I guess, would then result in an exception?

You could resolve this with `if flash[:subscription] && (subscription = @subscriptions[flash[:subscription]['id']])`

Reply:
Hmm...I don't think so, which makes it awkward to write a test for. This flash is only shown just after the subscription has been created. In theory, someone could unsubscribe from it in the small amount of time when the redirect is happening, but that's veering towards impossible. Can you think of another scenario?

Reply:
Hmm sure, I don't have a good example of a realistic scenario. It just feels like something where why risk a crash from my opinion - so an example of defensive coding thing rather than something to test. I don't mind too much if you want to leave the risk in but my suggestion would be to just implicitly check for it's existence before proceeding.

Reply:
OK. I've added another commit with a test for the subscription.